New implementation of the simplified UI

WoodyLetsCode opened the pull request:
This turns the original UI into a simplified UI instead of having a second simplified UI. Now in the simple ui there is so much duplicate code and I think this would be a better implementation of the simplified ui. So my idea was that when the simple ui is enabled, we fetch the default ui, hide all unnecessary ui elements and then show the ui.

This is what it looks like:

![grafik](https://github.com/Aircoookie/WLED/assets/27882680/be7634fa-a237-454b-87c8-5d8b6d436b20)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/87363dd4-cb75-454f-82c8-c287a16012ec)

![grafik](https://github.com/Aircoookie/WLED/assets/27882680/4c78741e-1a00-4b09-a65d-03b23c9c2fe8)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/953c139b-ab46-451d-b259-1dfcdc3a05f6)

![grafik](https://github.com/Aircoookie/WLED/assets/27882680/687034a6-cd79-4a97-9236-b616d36e932a) | ![grafik](https://github.com/Aircoookie/WLED/assets/27882680/eb1fce4e-55e6-4699-b2bf-eb10d260dc35)
:-------------------------:|:-------------------------:
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/b3648c51-dfa0-432f-98b2-e282758de092) | ![grafik](https://github.com/Aircoookie/WLED/assets/27882680/294e4544-d830-4945-934e-69eaf3d1031a)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/31d337d0-43f6-460f-8ad6-d582d0715d97) |

I ended up not having a button to open the palettes or effects because it just adds an unnecessary click that you have to do to change your effect or palette. In the old simplified UI, you then had to scroll as well, so it makes no difference if you always have palettes and effects visible or not.

Now, because the simplified ui takes up much less space: **Should we make the simplified ui compiled in by default?**

blazoncek commented:
I was under the impression that you were about to remove `simple.htm` and accompanying files.

WoodyLetsCode commented:
I wanted to get rid of the duplicate code.

blazoncek commented:
The solution looks interesting but I see no point in having two files to maintain (if you override one's layout). Could you not put everything into `index.js` and only have decision made on some status in `Info` object? Admittedly it does not exist ATM, but that can be fixed. It will also make `wled_server.cpp` a bit cleaner.

Question: Shall we introduce a new class (i.e. "nosimple") that can be used to hide (or otherwise manipulate) elements not belonging into simplified user interface?

I have yet to test it but from your screenshots I cannot see simplified segments. Are they intentionally hidden? IMO if there is just one force it's opacity to 255 and state to on and you can hide it, but if there are 2 or more, display their power buttons and opacity (brightness) sliders.

WoodyLetsCode commented:
At first I also thought about putting everything in a file, but since there was no information in the `Info` object about whether the simplified UI was enabled or not, I changed my mind and did it this way. But since you would also prefer to have everything in one file I will add to the `Info` object whether the simplified UI is turned on or not and based on this info in `index.js` display the simplified UI or not.

I'm not sure whether the new class is absolutely necessary, because only display is actually set to `none`. But I'll just see during development whether it's needed or not.

The simplified segments are not implemented yet because I didn't quite understand what you meant, but now I do. I will add that.

blazoncek commented:
One more question: Could you condense palettes and effects into a searchable drop-down list?

WoodyLetsCode commented:
Yes, I can do that. 👍🏼

WoodyLetsCode commented:
I moved everything to `index.js` and implemented the simplified segments and added a drop-down button for the palettes.
Basically I did that what @blazoncek suggested here:
> ... Could you not put everything into `index.js` and only have decision made on some status in `Info` object? Admittedly it does not exist ATM, but that can be fixed. ...

Here are some screenshots of what it looks like now:
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/b631ea5f-3e33-40ed-b5db-ec8a83c11779)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/1f6dc58a-3a67-4a09-bc1e-75ee3e2438d7)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/30de3f96-fc1c-464b-9d26-57849bd0e44a)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/0953898b-1ade-497d-a3a4-333dcae41855)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/e5db9ae1-3c96-4b6d-aeef-7658d8852e92)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/bae72a4c-4083-485e-9a99-20768d828f2a)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/bbca50cb-8707-499d-8ed7-c786433c25f2)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/a03a6e26-a9fb-45ff-ae4a-121471ecee91) | ![grafik](https://github.com/Aircoookie/WLED/assets/27882680/166ebc21-f1fa-47ab-a533-9fdf333ac4e8)
:-------------------------:|:-------------------------:
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/76ef965a-2157-4043-afa6-8b19b9205801) | ![grafik](https://github.com/Aircoookie/WLED/assets/27882680/bef4abc8-de03-4e26-88d4-2ea6ded9b677)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/cb194512-bafa-452a-8078-db08196e1ac0) | ![grafik](https://github.com/Aircoookie/WLED/assets/27882680/9d051eae-a762-4a1d-b67d-07c13927e6a6)

I have two questions:
1. Do we really need an effects drop-down? I don't see the benefit of having one. Changing the effect is such a basic thing that you probably do every time you open the UI. And if we add a drop-down list for that, you would have to do an extra click every time you want to change the effect. Do you see this the same way?
2. In the current simple UI, the button that opens the drop-down menu has the name of the current palette/effect. I think this can be a bit confusing. For example, if you have selected the `Wavesins` effect, a less tech-savvy user might be confused as to what this button does. 
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/b9b50d8b-2e51-44f1-8bed-5f8ebb273acf)
So I changed the text of the button to `Change palette` / `Hide palettes`.
What do you think? Is that better?

I am open to suggestions for improvement and feedback. 😄

blazoncek commented:
1. I agree but I still think the list may be too long (180+ effects in total)
2. not if styling would be different IMO

Nevertheless it is nice to see this progress.
I can't wait to find the time to put it through my fingers. 😉

blazoncek commented:
@WoodyLetsCode ❤️ 

I finally had the time to testdrive this PR. Marvelous job!

Still, I'd do this (no segment names and thinner segment spacing):
![Screenshot 2023-11-22 at 15 55 23](https://github.com/Aircoookie/WLED/assets/7722392/882d7684-14b0-478d-a4e1-6ac232a9364e)

And I found this visual error (incorrect width of palettes):
![Screenshot 2023-11-22 at 15 46 50](https://github.com/Aircoookie/WLED/assets/7722392/b4835ab0-a95f-4ac6-8642-c7581da86b07)

I do miss presets (at least quick loads are there) but I get there needs to be some sort of compromise.

@Aircoookie and @softhack007 please review and comment.

WoodyLetsCode commented:
Thanks @blazoncek for testing!

I will change the segments according to your screenshot. 👍🏼 

I will also look into the visual error, I couldn't reproduce it at the moment.

I thought it would be better to just have the quick load of presets, because that would lead to a minimalized and simpler UI. But if you want to have presets in the simple UI, I could add that as well.

I'm currently torn between two design ideas. Which one do you prefer?
The text above shows the current palette:
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/1482f7b8-c75d-46cf-b297-266768f06443)
As it is in the current simple ui:
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/4b1e047d-cca8-4c00-a342-d20412a85f49)


Or do you have other ideas? I'm open to that as well.

blazoncek commented:
I prefer my solution, but I leave it to you.
Reasoning: Users (especially inexperienced) will get used to tap/click twice to get to palettes or effects. But presets are there for a reason (most likely created by someone else for those users) and should be accessed with a single tap/click.

Moustachauve commented:
I'd just like to give my opinion on the design of the palette selector. 
Based on the screenshots only, it looks weird. It just makes separated boxes of palette appear in the nornal page?

I would have two alternatives suggestions
 1) Make it look like a dropdown menu, with a different background and no space between the elements. I would also drop the round borders
2) Show the palette in a "dialog" on top of the whole user interface with a title "Select a palette" and a button "cancel" at the bottom.

blazoncek commented:
@Moustachauve that would be preferable but would also mean (I think) to make two versions of UI (as was the case with simple.htm).
What @WoodyLetsCode is trying to achieve is to modify existing UI with CSS and a bit of JS to be displayed in a simplified way.

Moustachauve commented:
It could be done by switching some classes in JS I believe.

blazoncek commented:
> It could be done by switching some CSS classes in JS I believe.

Please help. 😄

blazoncek commented on wled00/data/index.js:
```diff
@@ -825,6 +826,8 @@ function populateSegments(s)
 
 	gId('segcont').innerHTML = cn;
+	if (simplifiedUI) gId('segcont').classList.add("simple");
+	else              gId('segcont').classList.remove("simple");
```
cool 👍

blazoncek commented on wled00/data/index.js:
```diff
@@ -825,6 +826,8 @@ function populateSegments(s)
 	}
 
 	gId('segcont').innerHTML = cn;
+	if (simplifiedUI) gId('segcont').classList.add("simple");
```
most likely.

Moustachauve commented:
I don't know how to contribue to this PR (I couldn't push), but I made this change here to turn the palette selection into a dialog:
https://github.com/Moustachauve/WLED/commit/a389e41e03108406aff4d787595994d13eee1a65

It's an early prototype that could probably be improved. Here's what it looks like:
![image](https://github.com/Aircoookie/WLED/assets/2206577/3fe1fc51-b9fe-4a90-9d13-6a1a4c912b39)

blazoncek commented:
> I don't know how to contribue to this PR (I couldn't push),

Add a new remote (Woody's WLED.git) to your fork (`git remote add WoodyLetsCode https://github.com/WoodyLetsCode/WLED.git`), checkout his branch (`git checkout -b simple-mode WoodyLetsCode/simple-mode`), make changes, push to your fork (`git push origin`), make a PR to Woody's fork. It is much easier to do that mumbo-jumbo using Git client in VSC.

WoodyLetsCode commented:
Thank you @Moustachauve ! ❤️ 

Your screenshot looks really impressive! I can't wait to get a PR from you.

WoodyLetsCode commented:
Good idea. I gave @Moustachauve collaborator rights

Moustachauve commented:
I made PR [#2 i](https://github.com/WoodyLetsCode/WLED/pull/2) in your repo

WoodyLetsCode commented:
Thank you @Moustachauve !
I took a look at it and there seems to be something wrong.

WoodyLetsCode commented:
@blazoncek Do you want to have this in `0_14_1` or `0_15` branch. I'm not sure because @softhack007 added this to the [0.14.2 candidate](https://github.com/Aircoookie/WLED/milestone/17) milestone.

WoodyLetsCode commented:
Okay, I will rebase it. 👍🏼

WoodyLetsCode commented:
I also added the dropdown menu for effects.
I'm still not sure which I like better. 🤔  The implementation with or without dropdown for the effects. Without dropdown, you need one click less to set the effect. But it looks cleaner with dropdown. 🤷🏼 
But I'm going to keep it that way, because @blazoncek  prefers it that way.

blazoncek commented:
One more thing. Buttons for pixel art, and custom palettes (add/delete) can be hidden too.

blazoncek commented:
Tested and approved.
@softhack007 and @Aircoookie please give it a try.

@WoodyLetsCode the scroll bar in a popup window for effects or palettes still cannot be accessed using mouse in Firefox. Scrolling with mouse wheel does work though. I have not yet tested it on mobile so it may need further tweaking.

WoodyLetsCode commented:
@blazoncek Can you tell me which version of Firefox you are using and which operating system you use? On my Firefox it works after my commit 930a1ae450a7e86fcf100e253ca11cb21c5cde8d

blazoncek commented:
Using FF 115.5.0esr on macOS 10.14

WoodyLetsCode commented:
It seems that there is a difference between Firefox for Windows and Firefox for MacOS.
I need to find a solution for this.

WoodyLetsCode commented:
@blazoncek It should work on macOS now.

I also made a few other small changes:
The background of the dialog is now blurred instead of darkened.
This is how it looks now:
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/2996179e-733b-4eed-a439-3495c40d8cce)
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/e32eaf29-b087-4e03-923f-7394acd97020)

Moustachauve commented:
@WoodyLetsCode Blur looks good on a photo! I wonder what does it look like on the default single color background? Does it have enough contrast?

WoodyLetsCode commented:
@Moustachauve  This is how it looks like on the default single color background:
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/4eef0764-7d19-4b81-b79b-93a3c170cb41)

Here in light mode:
![grafik](https://github.com/Aircoookie/WLED/assets/27882680/46952c5c-91ab-4206-ba46-244d64aaf392)
